README.md

```diff
@@ -115,6 +115,21 @@ You need to sign up for a free account, and also deal with configuring the GitHu
       * Secret: Enter a shared secret (some longish random text)
       * Events: "Send me everything"
 
+## Tests
```

Comment:
@ehuss by reading this README snippet I don't easily understand at a glance how the testsuite is meant to be used and maintained. Is the documentation actually meant to be in `tests/github_client/mod.rs` as rustdoc's comments? 

A few questions ahead. Sorry for the perhaps vague comments :slightly_smiling_face:, I'm trying to squint and understand these changes and their mainteinance cost for future other contributors:

- What is needed to test a new handler under `./src/handlers` (which is something I'm currently doing)? I assume little to nothing if the new handler is just using the same Github API already tested, right?

- (echoing the comment from Mark) What is needed to do to update tests? Say, we change a little something in how we interpret Github responses in some structs under `./src/github.rs`? 

- is there a way to regenerate these JSON files automatically?

Reply:
> Is the documentation actually meant to be in tests/github_client/mod.rs as rustdoc's comments?

I'm not quite sure I understand the question. The documentation in README.md is intended as a high-level overview of testing of triagebot, and includes a link to `tests/github_client/mod.rs` to read more details of that specific test suite. If and when more suites are added, they could be linked here, too.

> What is needed to test a new handler under ./src/handlers (which is something I'm currently doing)? I assume little to nothing if the new handler is just using the same Github API already tested, right?

Right, handler testing isn't part of this PR. Testing of handlers is implemented in #1678, but I'm finding them to be even harder to wield than these tests, so I broke out just the github side of things to try to make some progress. 

> What is needed to do to update tests? Say, we change a little something in how we interpret Github responses in some structs under ./src/github.rs?

As mentioned above. The general process is to point the test at one of your personal repos, and re-record the live data.

> is there a way to regenerate these JSON files automatically?

Also mentioned above. I can take a look at making it easier to regenerate them.